### ISSUE TITLE
update all compose libraries to support compose 1.4.0

### ISSUE THREAD

**Issue by sunildhiman90** [adrielcafe/voyager#145]
Hi, i tried using voyager in my compose multi platform project, But i m facing following error there

```
Unbound public symbol IrSimpleFunctionPublicSymbolImp error in iOS build
```
check following issue link for more: https://github.com/JetBrains/compose-multiplatform/issues/3175

And in one issue they are saying that this issue is resolved in compose 1.4.0, and kotlin 1.8.20
check this issue for reference:  https://github.com/JetBrains/compose-multiplatform/issues/2903#issuecomment-1504895233

Then i checked out in voyager 1.0.0-rc06 , it is using following versions i think , let me know if am wrong

```
composeAnimation = "1.3.3"
composeFoundation = "1.3.1"
composeMaterial = "1.3.1"
composeRuntime = "1.3.3"
composeUI = "1.3.3"
```

So i want to know when we will be upgrading following compose dependencies version to 1.4.0 as well in voyager so that i will be completely compatible with compose 1.4.0.
OR 
let me know i am missing anything and provided wrong data here

Attachment for reference
<img width="1420" alt="Screenshot 2023-05-17 at 1 13 13 PM" src="https://github.com/adrielcafe/voyager/assets/35147222/65e67874-d155-48aa-910d-fd5261519376">

Thanks

**Comment by Syer10** [comment]
Those versions are only used by the samples, everything else uses the gradle plugin's version

**Comment by sunildhiman90** [comment]
Ok @Syer10 , thnx, then what can be the reason for that error, related to this ticket:
https://github.com/JetBrains/compose-multiplatform/issues/3175

do you have any idea?

**Comment by TheMrDOC111** [comment]
I have same problem

**Comment by sunildhiman90** [comment]
Hi @Syer10 is there any workaround available for this issue with voyager integrations in ios build, 
 
```
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.navigator.tab/Tab.icon.<get-icon>|-5764684600227404327[0]
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.navigator.tab/Tab.options.<get-options>|7766868889533212780[0]
Unbound public symbol IrSimpleFunctionPublicSymbolImpl: cafe.adriel.voyager.navigator.tab/Tab.title.<get-title>|371973571756195065[0]

This could happen if there are two libraries, where one library was compiled against the different version of the other library than the one currently used in the project. Please check that the project configuration is correct and has consistent versions of dependencies.
error: Compilation finished with errors
```

I am stuck with this issue, any help will be appreciated.

**Comment by sunildhiman90** [comment]
Closing this, as this issue is from compose side, We can do one workaround for this for now till this fix is available in compose. 
workaround is available here: #3175(comment)
